src/main/java/studybuddy/commands/EditCommand.java

```diff
@@ -0,0 +1,109 @@
+package studybuddy.commands;
+
+import studybuddy.CEGStudyBuddy;
+import studybuddy.course.Course;
+
+public class EditCommand extends Command {
+
+    public EditCommand(String param) {
+        super(param);
+    }
+
+    public String[] parseEdit() {
+        assert (!param.isEmpty());
+
+        String[] parts = param.split(" ");
+        String code = null;
+        String title = "";
+        Integer mc = null;
+        Integer y = null;
+        Integer s = null;
+
+        for (int i = 0; i < parts.length; i++) {
+            if (parts[i].startsWith("c/")) {
+                code = parts[i].substring(2);
+            } else if (parts[i].startsWith("t/")) {
+                // title may consist of a few words
+                title = parts[i].substring(2);
+                i += 1;
+                // while i is in bound and parts[i] is not next identifier
+                while (i < parts.length && !hasIdentifier(parts[i])) {
+                    title += " " + parts[i];
+                    i += 1;
+                }
+                // reset i back to the ending element of title
```

Comment:
Avoid comments for your own/coder's reference. Write comments for the reader's reference to maximize readability.